apis/v1beta1/targetallocator_types.go

```diff
@@ -36,24 +36,48 @@ type TargetAllocatorPrometheusCR struct {
 	// label selector matches no objects.
 	// +optional
 	PodMonitorSelector *metav1.LabelSelector `json:"podMonitorSelector,omitempty"`
+	// Namespaces to be selected to look for PodMonitors for target discovery.
+	// A label selector is a label query over a set of resources. The result of matchLabels and
+	// matchExpressions are ANDed. An empty label selector matches all objects. A null
+	// label selector matches no objects.
+	// +optional
+	PodMonitorNamespaceSelector *metav1.LabelSelector `json:"podMonitorNamespaceSelector,omitempty"`
```

Comment:
I wonder if we should also do some defaulting here to prevent an upgrade from breaking existing installations? As the doc string states, a null selector matches nothing, upgrading with this on will mean that existing installations (who do _not_ set this currently) would not have it, resulting in a null selector, aka :boom:

Reply:
Great point 👍 , should we implement that for the `PodMonitorNamespaceSelector ` or for all the selectors (including the already existing ones) ? What do you recommend ?

Reply:
Hey @swiatekm , thank you for the heads up for the kubebuilder 😃  I indeed found a way to set a default and I added it to the new fields like so 
```go
// +kubebuilder:default:={}
```
which resulted in a generate crd like this: 
```yaml
...
                      podMonitorNamespaceSelector:
                        default: {}
                        properties:
                          matchExpressions:
...
```
Let me know if that looks good for you 🙏